### ISSUE TITLE
Out of memory

### ISSUE THREAD

**Issue by LBierkandt** [mdaines/viz.js#72]
Using one of the v1.5.0 builds, I get an `uncaught exception: out of memory` when generating a number of graphs in a row.

**Comment by mdaines** [comment]
As of 1.5.0, Viz.js doesn't try to reuse the memory allocated on its initial run. This is an attempt to address situations like #65 and #59, but admittedly using kind of a blunt instrument.

What browser are you seeing this in? How many graphs are you generating in a row? Are they generated inside a `for` loop or in some other way? Can you post an example graph?

Both techniques I've used in the memory test appear to work in the browsers I've had a chance to test so far: https://github.com/mdaines/viz.js/blob/524f7f84cdb392f1b72e9981fdfe1dc3b71124c6/tests/memory.js

**Comment by LBierkandt** [comment]
Thank you for the quick reaction!

I'm using Firefox 47.0.2 on Windows. The graphs are not created in a loop but manually when browsing through the [app](https://github.com/lbierkandt/graph-anno)'s sentences (it's an app for linguistically annotated text). The memory exception occurs after rendering about 60 graphs. An example graph:
```
digraph G{
	type="digraph";rankdir="TB";use="dot";ranksep="0.3";
	949[fontname="Helvetica",color="#000000",shape="box",label="S&#xa;n0",fontcolor="#000000",]
	950[fontname="Helvetica",color="#000000",shape="box",label="S&#xa;n1",fontcolor="#000000",]
	951[fontname="Helvetica",color="#00cc00",shape="box",label="R&#xa;n2",fontcolor="#00cc00",]
	949->944[fontname="Helvetica",color="#000000",weight="10",constraint="true",label="s&#xa;e0",fontcolor="#000000",]
	951->944[fontname="Helvetica",color="#00cc00",weight="1",constraint="true",label="ex&#xa;e1",fontcolor="#00cc00",]
	949->945[fontname="Helvetica",color="#000000",weight="10",constraint="true",label="pr&#xa;e2",fontcolor="#000000",]
	950->946[fontname="Helvetica",color="#000000",weight="10",constraint="true",label="aux&#xa;e3",fontcolor="#000000",]
	950->947[fontname="Helvetica",color="#000000",weight="10",constraint="true",label="s&#xa;e4",fontcolor="#000000",]
	951->947[fontname="Helvetica",color="#00cc00",weight="1",constraint="true",label="ex&#xa;e5",fontcolor="#00cc00",]
	950->948[fontname="Helvetica",color="#000000",weight="10",constraint="true",label="pr&#xa;e6",fontcolor="#000000",]
	949->950[fontname="Helvetica",color="#00cc00",weight="1",constraint="true",label="ad&#xa;rel: caus&#xa;e7",fontcolor="#00cc00",]
	944->945[style="invis",weight="100",]945->946[style="invis",weight="100",]946->947[style="invis",weight="100",]
	947->948[style="invis",weight="100",]
	subgraph wnabyquvkgfmjxes{
		rank="same";
		944[fontname="Helvetica",label="er&#xa;t0",shape="box",style="bold",color="#000000",fontcolor="#000000",]
		945[fontname="Helvetica",label="stirbt&#xa;t1",shape="box",style="bold",color="#000000",fontcolor="#000000",]
		946[fontname="Helvetica",label="weil&#xa;t2",shape="box",style="bold",color="#000000",fontcolor="#000000",]
		947[fontname="Helvetica",label="er&#xa;t3",shape="box",style="bold",color="#000000",fontcolor="#000000",]
		948[fontname="Helvetica",label="lacht&#xa;t4",shape="box",style="bold",color="#000000",fontcolor="#000000",]
	}
	subgraph vpmznchgjtuxbrfe{
		949[]
		950[]
	}
	subgraph ohzxavtqesiunwlk{
		951[]
	}
}
```

**Comment by mdaines** [comment]
I was able to get Firefox 47 on Windows to throw an out of memory error using the attached test case at about ~200 calls for that graph (see console for counts). However, Firefox 50 passes that test (1000 calls).

[memory-test.zip](https://github.com/mdaines/viz.js/files/714075/memory-test.zip)

**Comment by LBierkandt** [comment]
I'm very sorry for the late response – I've been too busy with other stuff...

The memory issue seems to be quite system-dependent. When testing with Firefox 47 on Windows, 32bit, 3GB RAM, the error occured after 70 graphs. I also checked with FF 51 on Linux (same hardware) – there the error occured after generating 60 graphs (though, surprisingly, it ran up to 550 once).

I also tested v1.7.0 with totalMemory set to 32MB, which didn't throw the error (32 was the highest value possible). Next I'll test the new version on my app...
